server/controllers/chatController.js

```diff
@@ -0,0 +1,221 @@
+          { $group: { _id: { profileId: "$profile", userId: "$_id" } } },
+        ]);
+        //find Profile Data for every companion
+        const profileIds = userProfiles.map(item => {
+          return mongoose.Types.ObjectId(item._id.profileId);
+        });
+        let profilesData = await Profile.aggregate([
+          { $match: { _id: { $in: profileIds } } },
+          {
+            $group: {
+              _id: {
+                firstName: "$firstName",
+                lastName: "$lastName",
+                profile: "$_id",
+                picture: "$profilePicture",
+              },
+            },
+          },
+        ]);
+        //Bring all the data together and create array of objects
+        let conversations = profilesData.map(item => {
+          return {
+            firstName: item._id.firstName,
+            lastName: item._id.lastName,
+            picture: item._id.picture,
+            profileId: item._id.profile,
+            userId: userProfiles.filter(element => {
+              return (
+                element._id.profileId.toString() == item._id.profile.toString()
+              );
+            })[0]._id.userId,
+          };
+        });
+        conversations.map(item => {
+          item.lastMessage = lastMessagesList.filter(element => {
+            return (
+              element.companionUserId.toString() === item.userId.toString()
+            );
+          })[0].lastMessage;
+          item.chatId = lastMessagesList.filter(element => {
+            return (
+              element.companionUserId.toString() === item.userId.toString()
+            );
+          })[0].chatId;
+        });
+        //sort chats the way latest message will be on top, and chats with no messages will be at the bottom
+        conversations.sort((a, b) => {
+          if (a.lastMessage) {
+            if (b.lastMessage) {
+              if (a.lastMessage.timeCreated > b.lastMessage.timeCreated) {
+                return -1;
+              } else {
+                return 1;
+              }
+            } else {
+              if (a.lastMessage.timeCreated > 0) {
+                return -1;
+              }
+            }
+          } else if (!a.lastMessage && !b.lastMessage) {
+            if (a.chatId > b.chatId) {
+              return -1;
+            }
+          }
+        });
+        res.json({ error: false, data: conversations });
+      }
+    );
+  } catch (err) {
+    next(createError(500, err.message));
+  }
+};
+
+exports.sendMessage = async (req, res, next) => {
+  try {
+    const { content, chatId } = req.body;
+    const sender = req.user.id;
+    await Chat.findById(chatId, async (err, chat) => {
+      if (err) console.log(err);
+      messageId = chat.messages;
+      Message.findById(messageId, (err, messages) => {
+        if (err) console.log(err);
+        messages.messages.push({
+          sender: sender,
+          content: content,
+          timeCreated: moment().format(),
```

Comment:
I don't have an issue with this approach, but you could also do this automatically by setting `timestamp: true` (I believe that's the syntax) in the Schema definition